### PR TITLE
DBZ-7221 refactored mongo sharded tests

### PR DESCRIPTION
**Pull request by MartinMedek**

### CODE REVIEW

File: debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/fixtures/databases/ocp/OcpMongoSharded.java [hunk 1]
```diff
@@ -33,14 +28,23 @@ public OcpMongoSharded(ExtensionContext.Store store) {
...
-        OcpMongoShardedDeployer deployer = new OcpMongoShardedDeployer.Deployer()
-                .withOcpClient(ocp)
+        OcpMongoShardedDeployer deployer = OcpMongoShardedDeployer.builder()
                 .withProject(ConfigProperties.OCP_PROJECT_MONGO)
-                .withMongosDeployment(MONGOS_DEPLOYMENT)
-                .withConfigDeployment(CONFIG_DEPLOYMENT)
-                .withServices(SERVICES)
+                .withOcp(ocp)
+                .withConfigServerCount(OcpMongoShardedConstants.CONFIG_SERVER_REPLICAS)
+                .withShardCount(OcpMongoShardedConstants.SHARD_COUNT)
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/fixtures/databases/ocp/OcpMongoSharded.java:35]: You are passing parameters but you are not using it.

**Review by obabec — CHANGES_REQUESTED** [review]

**Review by jcechace — COMMENTED** [review]
Few comments mostly around the ability to understand when reading the code. I'm also missing javadocs

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mongodb/sharded/OcpMongoShardedReplicaSet.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+import io.debezium.testing.system.tools.databases.mongodb.sharded.freemarkermodels.FreemarkerConfiguration;
+import io.debezium.testing.system.tools.databases.mongodb.sharded.freemarkermodels.InitReplicaSetModel;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import lombok.Builder;
+import lombok.Getter;
+
+public class OcpMongoShardedReplicaSet implements Startable {
```
[the remark this answers is not in the document]
Comment (MartinMedek) [reply debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mongodb/sharded/OcpMongoShardedReplicaSet.java]: It is replica set in ocp sharded mongo cluster. I changed the naming and hope it's more clear.

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mongodb/sharded/MongoShardedUtil.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        MongoShardKey customersKey = new MongoShardKey("inventory.customers", "_id", MongoShardKey.ShardingType.NORMAL);
+        customersKey.getKeyRanges().add(new ShardKeyRange(OcpShardModelProvider.getShardReplicaSetName(1), "1000", "1003"));
+        customersKey.getKeyRanges().add(new ShardKeyRange(OcpShardModelProvider.getShardReplicaSetName(2), "1003", "1004"));
+
+        MongoShardKey productsKey = new MongoShardKey("inventory.products", "_id", MongoShardKey.ShardingType.HASHED);
+        return List.of(customersKey, productsKey);
+    }
+
+    public static String createRootUserCommand(String userName, String password) {
+        return "db.getSiblingDB('admin').createUser({user: '" + userName + "', pwd: '" + password + "', roles: [{role:\"root\",db:\"admin\"}] })";
```
[the remark this answers is not in the document]
Comment (MartinMedek) [reply debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mongodb/sharded/MongoShardedUtil.java:60]: Good tip, I shall investigate for future what can be replaced with the client.

**Review by obabec — APPROVED** [review]

**Pull request merged**